jboss/container/wildfly/launch/https/added/https.sh

```diff
@@ -97,19 +100,56 @@ function configureHttpsXml() {
-    local ret
-    testXpathExpression "${xpath}" "ret"
+    local ssRet
+    testXpathExpression "${xpath}" "ssRet"
+    if [ "${ssRet}" -ne 0 ]; then
+      echo "You have set HTTPS_PASSWORD, HTTPS_KEYSTORE_DIR and HTTPS_KEYSTORE to add an undertow https-listener. Fix your configuration to contain the undertow subsystem for this to happen." >> "${CLI_SCRIPT_ERROR_FILE}"
+      return
+    fi
+
+    # Not having any servers is an error
+    local serverNamesRet
+    # We grab the <server name="..."> attributes, and will use them later
+    local xpath="\"//*[local-name()='subsystem' and starts-with(namespace-uri(), 'urn:jboss:domain:undertow:')]/*[local-name()='server']/@name\""
+    testXpathExpression "${xpath}" "serverNamesRet"
+    if [ "${serverNamesRet}" -ne 0 ]; then
+      echo "You have set HTTPS_PASSWORD, HTTPS_KEYSTORE_DIR and HTTPS_KEYSTORE to add an undertow https-listener. Fix your configuration to contain at least one server in the undertow subsystem for this to happen." >> ${CLI_SCRIPT_ERROR_FILE}
+      return
+    fi
+
+    # Existing https-listener(s) is an error
+    local httpsListenersRet
+    local xpath="\"//*[local-name()='subsystem' and starts-with(namespace-uri(), 'urn:jboss:domain:undertow:')]/*[local-name()='server']/*[local-name()='https-listener']/@name\""
+    testXpathExpression "${xpath}" "httpsListenersRet"
+    if [ "${httpsListenersRet}" -eq 0 ]; then
+      echo "You have set HTTPS_PASSWORD, HTTPS_KEYSTORE_DIR and HTTPS_KEYSTORE to add https-listeners to your undertow servers, however at least one of these already contains an https-listener. Fix your configuration." >> "${CLI_SCRIPT_ERROR_FILE}"
+      return
+    fi
 
-    if [ "${ret}" -eq 0 ]; then
     cat << EOF >> ${CLI_SCRIPT_FILE}
     for serverName in /subsystem=undertow:read-children-names(child-type=server)
-        if (result == []) of /subsystem=undertow/server=\$serverName:read-children-names(child-type=https-listener)
-          /subsystem=undertow/server=\$serverName/https-listener=https:add(security-realm=ApplicationRealm, socket-binding=https, proxy-address-forwarding=true)
-        else
-          echo There is already an undertow https-listener for the '\$serverName' server so we are not adding it >> \${warning_file}
-        end-if
+      /subsystem=undertow/server=\$serverName/https-listener=https:add(security-realm=ApplicationRealm, socket-binding=https, proxy-address-forwarding=true)
     done
 EOF
-    fi
+
+}
+
+# An XPath expression e.g getting all name attributes for all the servers in the undertow subsystem
+# will return a variable with all the attributes with their names on one line, e.g
+#     'name="server-one" name="server-two" name="server-three"'
+# Call this with ($input is the string above)
+#     convertAttributesToValueOnEachLine "$input" "name"
+# to convert this to :
+# "server-one
+# server-two
+# server-three"
+function splitAttributesStringIntoLinks() {
```

Comment:
Is this function used somewhere?

Reply:
No (I was using it but didn't need it) - removed